packages/platform-android/src/commands/runAndroid/runOnEmulator.ts

```diff
@@ -26,29 +26,62 @@ function toPascalCase(value: string) {
 
 type AndroidProject = NonNullable<Config['project']['android']>;
 
-async function runOnAllDevices(
+function logStartEmulatorError(error: Error) {
+  if (error.message) {
+    logger.error(
+      `Failed to launch emulator. Reason: ${chalk.dim(error.message)}`,
+    );
+  } else {
+    logger.error('Failed to launch emulator.');
+  }
+
+  logger.warn(
+    'Please launch an emulator manually or connect a device. Otherwise the app may fail to launch.',
+  );
+}
+
+async function startEmulator(adbPath: string, deviceName?: string) {
+  if (deviceName) {
+    logger.info('Launching emulator...');
+
+    try {
+      const deviceListAfterEmulatorLaunched = await launchEmulator(
+        deviceName,
+        adbPath,
+      );
+
+      return deviceListAfterEmulatorLaunched;
+    } catch (error) {
+      return logStartEmulatorError(error);
+    }
+  }
+
+  const devices = adb.getDevices(adbPath);
+  if (devices.length > 0) {
+    return devices;
+  }
+
+  logger.info('Launching emulator...');
+
+  try {
+    const deviceListAfterAnyEmulatorLaunched = await launchAnyEmulator(adbPath);
+
+    return deviceListAfterAnyEmulatorLaunched;
+  } catch (error) {
+    console.log({error});
```

Comment:
This `console.log` is extraneous here. There's already logging inside `logStartEmulatorError`.